src/activitys/ChallengeModeDungeon.ts

```diff
@@ -0,0 +1,158 @@
+import { Metadata } from "main/types";
+import { dungeonTimersByMapId, VideoCategory } from "../main/constants";
+import { ChallengeModeTimelineSegment, TimelineSegmentType } from "../main/keystone";
+import Activity from "./Activity";
+
+export default class ChallengeModeDungeon extends Activity {
+    private mapID: number;
+    private level: number;
+    private affixes: number[];
+    private timings: number[];
+    private CMDuration: number = 0;
+    private chests?: number;
```

Comment:
Better to name this `upgradeLevel`. "chests" is not an official term for anything. =D